webapp/lib/controller.dart

```diff
@@ -90,6 +102,61 @@ void initUI() {
   );
 }
 
+void setupProjectTimers() {
+  var allProjects = view.contentView.projectSelectorView.allProjects;
+  for (var project in allProjects) {
+    var timer = new Timer(Duration(hours: 1), () => view.contentView.stale = true);
+    projectTimers.add(new ProjectTimer(project, timer));
+  }
+}
+
+void setupProjectTimer(String project) {
+  var timer = new Timer(Duration(hours: 1), () => view.contentView.stale = true);
+  projectTimers.add(new ProjectTimer(project, timer));
+}
+
+void checkNeedsReplyMetricsStale([bool isProjectSelection = false]) {
+  List<model.NeedsReplyData> sortedNeedsReplyDataList = List.from(needsReplyDataList);
+  sortedNeedsReplyDataList.sort((d1, d2) => d1.datetime.compareTo(d2.datetime));
+  var allProjects = view.contentView.projectSelectorView.allProjects;
+
+  List<model.NeedsReplyData> latestDataPerProject = [];
+  for (var project in allProjects) {
+    var latestProject = sortedNeedsReplyDataList.where((entry) => entry.project == project).last;
+    latestDataPerProject.add(latestProject);
+  }
+
+  if (projectTimers.length < 1) {
+      setupProjectTimers();
+  } else {
+    for (var entry in latestDataPerProject) {
+      var projectTimer = projectTimers.firstWhere((projectTimer) => projectTimer.project == entry.project);
+      if (projectTimer != null) {
+        projectTimer.timer.cancel();
+        setupProjectTimer(entry.project);
```

Comment:
Maybe I'm misunderstanding, but this seems to create a 1h timer for _all_ projects whenever new data comes in, even if it's not just the project that has been updated... We should be updating the timer only for those projects that have got new data

Reply:
timer now created only for the selected project